pytest/tests/sanity/epoch_switches.py

```diff
@@ -1,9 +1,11 @@
 #!/usr/bin/env python3
 # Spins up four nodes, and alternates [test1, test2] and [test3, test4] as block producers every epoch
 # Makes sure that before the epoch switch each block is signed by all four
+# We are not focusing on state sync in this test, so the nodes will continuously monitor the shard.
```

Comment:
That's interesting case. I guess it's hard to support because there is no consistent source of state parts, right? We could add separate dumper (observer) node, but it looks unnecessary for the test then, as you said.

Reply:
We already have nayduck tests for state sync. I think that it would add complexity to the test if we make the nodes state sync.

I can also see the benefit of looking at these features along with state sync as well.